boards/common/openocd.board.cmake

```diff
@@ -13,6 +13,26 @@ else()
   set_ifndef(OPENOCD_FLASH "flash write_image erase")
 endif()
 
+# SoC specific mass erase options
```

Comment:
Doesn't feel like the right place to add this in the generic openocd.board.cmake? Maybe have an openocd-stm32.board.cmake similar to openocd-nrf5.board.cmake or something?

Reply:
Thats fair feedback, what I was trying to avoid was an intrusive PR that touches each of the >100 ST boards in tree, and doesn't add the feature for any out of tree boards. Could an acceptable middle ground be moving the content to a `stm32.board.cmake` file, but including it directly from `openocd.board.cmake`?

Reply:
I'd prefer to keep it aligned with nrf, and add an `openocd-stm32.board.cmake`, but not a strong opinion here.

Reply:
My issue with including `openocd-stm32.board.cmake` from stm32 boards is that we'll be calling
```
include(${ZEPHYR_BASE}/boards/common/openocd-stm32.board.cmake)
```
instead of following one liner (to be tuned depending on target)
```
board_runner_args(openocd "--cmd-erase=stm32l1x mass_erase 0")
```

So it doesn't add much value except the fact that you don't have to tune depending on the board and then it can be added "blindly" to any stm32 based board. So, I'd rather be in favor of current proposal, which makes things fully transparent to user.

But not a strong opinion either.

Reply:
Maybe a small improvement is to have this in `openocd-stm32.board.cmake`, but include it in this file to declutter it a bit?

```cmake
include(${CMAKE_CURRENT_LIST_DIR}/openocd-stm32.board.cmake)
```

Reply:
> Maybe a small improvement is to have this in `openocd-stm32.board.cmake`, but include it in this file to declutter it a bit?
> 
> ```cmake
> include(${CMAKE_CURRENT_LIST_DIR}/openocd-stm32.board.cmake)
> ```

Sure, this would work for me.